common/src/main/java/com/viaversion/viabackwards/protocol/protocol1_17to1_17_1/Protocol1_17To1_17_1.java

```diff
@@ -91,7 +92,14 @@ public void registerMap() {
                     wrapper.write(Type.FLAT_VAR_INT_ITEM_ARRAY, wrapper.read(Type.FLAT_VAR_INT_ITEM_ARRAY_VAR_INT));
 
                     // Carried item - should work without adding it to the array above
-                    wrapper.read(Type.FLAT_VAR_INT_ITEM);
+                    Item carried = wrapper.read(Type.FLAT_VAR_INT_ITEM);
+                    if (carried != null) {
+                        // For click drag ghost item fix -- since the state ID is always wrong,
+                        // the server always resends the entire window contents after a drag action,
+                        // which is useful since we need to update the carried item in preparation
+                        // for a subsequent drag
+                        wrapper.user().get(PlayerLastCursorItem.class).setLastCursorItem(carried);
```

Comment:
You have to check if the PlayerLastCursorItem object is null (when the player is a 1.17.0 client). And shouldn't this also set the last cursor item if it is null?